usr/share/rear/lib/linux-functions.sh

```diff
@@ -168,15 +168,42 @@ function RequiredSharedObjects () {
+    #  1. Line: "/path/to/binary:"                                 -> current file argument for ldd
+    #  2. Line: "        lib (mem-addr)"                           -> virtual library
+    #  3. Line: "        lib => not found"                         -> print error to stderr
+    #  4. Line: "        lib => /path/to/lib (mem-addr)"           -> print $3 '/path/to/lib'
     #  5. Line: "        /path/to/lib => /path/to/lib2 (mem-addr)" -> print $3 '/path/to/lib2'
-    #  6. Line: "        /path/to/lib (mem-addr)"        -> print $1 '/path/to/lib'
-    ldd "$@" | awk ' /^\t.+ => not found/ { print "Shared object " $1 " not found" > "/dev/stderr" }
-                     /^\t.+ => \// { print $3 }
-                     /^\t\// && !/ => / { print $1 } ' | sort -u
+    #  6. Line: "        /path/to/lib (mem-addr)"                  -> print $1 '/path/to/lib'
+    local file_for_ldd=""
+    local file_owner_name=""
+    # It is crucial to append to /dev/$DISPENSABLE_OUTPUT_DEV (cf. 'Print' in lib/_input-output-functions.sh):
+    for file_for_ldd in $@ ; do
+        # Skip non-regular files like directories, device files, and non-existent files
+        # cf. similar code in build/GNU/Linux/100_copy_as_is.sh
+        # but here symbolic links must not be skipped (e.g. /sbin/mkfs.ext2 -> /usr/sbin/mkfs.ext2)
+        # otherwise there would be binaries in the recovery system without required libraries:
+        test -f "$file_for_ldd" || continue
+        # Skip the ldd test for kernel modules and firmware files
+        # which could happen via COPY_AS_IS+=( /lib/firmware/my_hardware )
+        # cf. the code in build/default/990_verify_rootfs.sh
+        egrep -q '/lib/modules/|/lib.*/firmware/' <<<"$file_for_ldd" && continue
+        # Skip files that are not owned by a trusted user to mitigate possible ldd security issues
+        # because some versions of ldd may directly execute the file (see "man ldd")
+        # which could lead to the execution of arbitrary programs as user 'root'
+        # in particular when directories are specified in COPY_AS_IS that may contain
+        # unexpected files like programs from arbitrary (possibly untrusted) users
+        # like COPY_AS_IS+=( /home/JohnDoe ) when JohnDoe is not a trusted user.
+        if test "$TRUSTED_FILE_OWNERS" ; then
+            file_owner_name="$( stat -c %U $file_for_ldd )"
+            if ! IsInArray "$file_owner_name" "${TRUSTED_FILE_OWNERS[@]}" ; then
+                Log "RequiredSharedObjects: Skipping 'ldd' for '$file_for_ldd' (owner '$file_owner_name' not in TRUSTED_FILE_OWNERS)"
+                continue
+            fi
+        fi
+        ldd $file_for_ldd | awk ' /^\t.+ => not found/ { print "Shared object " $1 " not found" > "/dev/stderr" }
+                                  /^\t.+ => \// { print $3 }
+                                  /^\t\// && !/ => / { print $1 } ' | sort -u
+    done 2>>/dev/$DISPENSABLE_OUTPUT_DEV
```

Comment:
Seems to me that the filter `| awk ... | sort -u` should go between `done` and `2>>/dev/$DISPENSABLE_OUTPUT_DEV` in the following line. Reasons:
* Only one subprocess invocation per entire loop.
* Otherwise there might be lots of duplicates in the output as `sort -u` does not cover the entire output set.

Reply:
Oops -  I overlooked that comment.
I will have a look...

Reply:
Fixed via
https://github.com/rear/rear/pull/2286/commits/76f1aa1e084bb0ba408f0edb9338c6722534721b

@OliverO2 
thank you so much for your careful review!
You helped a lot here to avoid a poor implementation regression
(the poor implementation even worked - at least for me - but poorly)
that I would have introduced because of my too narrow "focus of mind".
I concentrated too much on my primary goal and did not sufficiently
care about how the current code was meant to work.

With the fix the new code now still behaves as the old one was meant to work
and the "rear -D mkrescue" log file size decreased noticeably (by about 140 KiB).
Where it decreased is excatly a place where RequiredSharedObjects is called.
Without the fix the longest line with about 50000 characters in the log is
```
... Libraries being copied: [long list of libraries]
```
with many duplicate libraries but with the fix it is only about 6000 characters.

The matching code is in build/GNU/Linux/390_copy_binaries_libraries.sh
```
local all_libs=( "${LIBS[@]}" $( RequiredSharedObjects "${all_binaries[@]}" "${LIBS[@]}" ) )

Log "Libraries being copied: ${all_libs[@]}"
```
https://github.com/rear/rear/blob/master/usr/share/rear/build/GNU/Linux/390_copy_binaries_libraries.sh#L75

With `set -x` the line appears three times in the log which results
about 3*50000 - 3*6000 = 132000 less characters in the log.

Without the fix build/GNU/Linux/390_copy_binaries_libraries.sh
runs for about 16 secunds (for me)
but with the fix it needs only about 5 seconds
so "rear mkrescue" is about 11 seconds faster because of this fix.

@OliverO2 
again thank you for your careful review!